apis/sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/BuiltInFunctionResourceIntegrationTest.java

```diff
@@ -313,7 +320,112 @@ public void pushNoArray() {
           .then()
           .statusCode(400)
           .body("code", equalTo(400))
-          .body("description", equalTo("The path provided to push to has no array, found {}."));
+          .body(
+              "description",
+              equalTo(
+                  "The path provided to push to has no array, found {\"a\":3,\"b\":[{},{\"nested\":\"value\"},[]]}."));
+    }
+
+    @Test
+    public void setUpdateOneValue() {
+      String setOperation = "{ \"b.[1].nested\": \"newvalue\" }";
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .contentType(ContentType.JSON)
+          .body(String.format(SET_PAYLOAD, setOperation))
+          .when()
+          .post(BASE_PATH + "/object/function", DEFAULT_NAMESPACE, DEFAULT_COLLECTION, DOCUMENT_ID)
+          .then()
+          .body("documentId", equalTo(DOCUMENT_ID))
+          .statusCode(200);
+
+      // get whole document and check the value
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .get(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION, DOCUMENT_ID)
+          .then()
+          .body("documentId", equalTo(DOCUMENT_ID))
+          .body("data", jsonPartEquals("object.b[1].nested", "newvalue"))
+          .statusCode(200);
+    }
+
+    @Test
+    public void setUpdateOneValueCreate() {
```

Comment:
To be on par with mongo, if you `set` a field that doesn't yet exist, it will get created for you.